Automattic/jetpack
# Core Compat: Site Environment

Fixes #11798

#### Changes proposed in this Pull Request:
* Uses `wp_get_environment_type` as the default value for the Status package `is_staging_site` value (by way of !== production is a staging site in how we handle them).
* Uses the `WP_LOCAL_DEV` that Core site health does check. This activates our "Development Mode".
* Renames "Development Mode" to "Offline Mode" to be clearer our intent of this mode.
* Deprecation and updates relevant for ^.

#### Jetpack product discussion
n/a

#### Does this pull request change what data or activity we track or use?
n/a

#### Todo required
If this lands, we need to update documentation. I'm okay doing this in a folllow-up PR or adding a new commit here if this is otherwise approved.

If we're happy with this, I'll coordinate with Quill.

#### Testing instructions:
* Ensure Offline mode works (try this on localhost without a dot in the domain, e.g. our docker without using ngrok/JT.
* On a JN site running on WP trunk on a connected Jetpack site, set a constant `define( 'WP_ENVIRONMENT_TYPE', 'staging' );` Does this activate staging mode?

#### Proposed changelog entry for your changes:
* WordPress 5.5: Utilize new wp_get_environment_type function to determing Staging sites.
* Rename Development Mode to Offline Mode to be clearer our intent for its use.
